### ISSUE TITLE
[GDPR] Offer option for non-cookie based session/client ID identifiers

### ISSUE THREAD

**Issue by rolebi** [DataDog/browser-sdk#323]
Hi,

One of your product advantage compared to Google Analytics, is that it is not tainted by marketing / advertising / campaign performances purposes and thus can be used as a pure analytic tool in regard of the European privacy regulations.

Sadly, there's no option to make it works without dumping first (and third party) cookies.

With a "non cookie based session / client identifier" option, it would be possible to only display an opt-out banner to the user instead of having to show them a Cookie Policy wall with "Accept" and "Reject" options.

When using your SDK on Single Page Applications, having a session ID generated in memory is fine and do the job most of the time.

Having the application generate the client ID for the SDK (usually based on the application own user's unique identifier) and pass it during the init process or via a dedicated method would also works nicely for all application where a user has to log-in in some way.

**Comment by bcaudan** [comment]
Hi @rolebi,

Thanks for the feedback, this is a really interesting feature. We have not planned to work on it yet but we will keep it in mind.

**Comment by jskrepnek** [comment]
This would also enable support in browser environments where cookies are not fully supported, i.e.: Cordova / Ionic.

**Comment by sean6bucks** [comment]
Just wanted to add a bit to this, I noticed when implementing the RUM browser library, in addition to the `__dd_s` cookie, datadog is also adding several third-party cookies for what I assume is their own tracking (looks like GA, Hotjar, etc.) appearing under the `datadog.com` domain (unlike dd_s which comes from our domain)? Is there anyway to disable/prevent these from being added to use the product? Again mostly for GDPR concern since it is possibly tracking data that we cannot control.

**Comment by bcaudan** [comment]
@sean6bucks browser SDKs only creates first party cookies.
My guess is that the third party cookies under `datadog.com` that you have noticed, have been created by the datadog application.
Do you confirm?

**Comment by sean6bucks** [comment]
@bcaudan Indeed, thank you for that. had a datadog dashboard open in a tucked away window that was regenerating these cookies. Indeed when i closed that only first-party domain cookies were set. Cheers. (edited my original comment to prevent future confusion for readers)

**Comment by CsabaSzabo** [comment]
@bcaudan Are there any plans to implement this in the near future?

**Comment by bcaudan** [comment]
Hello @CsabaSzabo,

It is still a topic that we consider from time to time but no timeline to share for now.

**Comment by amortemousque** [comment]
Hello @marano,
This is a topic we have in mind but still no timeline to share.

**Comment by betovelandia** [comment]
@amortemousque this is a blocker for us, is there any way to get some traction on this matter?
